spec/models/user_spec.rb

```diff
@@ -0,0 +1,139 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  let(:email) { 'robert@example.com' }
+  let(:first_name) { 'robert' }
+  let(:last_name) { 'rodriguez' }
+  let(:user) { create_user(email, first_name, last_name) }
+
+  def create_user(email, first_name, last_name)
+    User.new(email:, first_name:, last_name:)
+  end
+
+  let(:oauth_client) { OauthClient.create!(client_id: '123', client_name: 'Test App', redirect_uri: 'https://example.com/callback') }
+
+  describe 'Validations' do
+    context 'when attributes are valid' do
+      it 'creates a valid user' do
+        expect(user.valid?).to be_truthy
```

Comment:
RSpec has a [direct `be_valid` test](https://www.rubydoc.info/gems/rspec-rails/RSpec%2FRails%2FMatchers:be_valid) that you can just call directly on your models.
```ruby
expect(user).to be_valid
expect(user).not_to be_valid
```